src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,108 @@
+package duke;
+
+import java.io.IOException;
+
+public class Duke  {
+
+    private Storage storage;
+    private TaskList taskList;
+    private UserInterface userInterface;
+    private GuiUserInterface guiUserInterface;
+
+    /**
+     * Constructor for Duke.
+     *
+     * @param path Relative path in local storage to save data.
+     */
+    public Duke(String path) {
+        try {
+            this.taskList = new TaskList();
+            storage = new Storage(path, taskList);
+            userInterface = new UserInterface(taskList);
+            guiUserInterface = new GuiUserInterface(taskList);
+            storage.load();
+        } catch (IOException e) {
+            UserInterface.showLoadingError();
+            this.taskList = new TaskList();
+        }
+    }
+
+    public Duke() {
+    }
+
+    ;
+
+    public static void main(String[] args) throws DukeException {
+//        new Duke("data/tasks.txt").run();
+        new Duke("data/tasks.txt");
+    }
+
+//    public void run() throws DukeException {
```

Comment:
I think that you should delete this block of code if you do not intend on using it in future increments.